client/src/components/BoxDisplay.js

```diff
@@ -29,22 +27,45 @@ function BoxDisplay({ studentId, studentName }) {
         }
         console.log(data);
         setStudentDetail(data);
-        // setIsClicked(!isClicked);
       });
   };
+
   useEffect(fetchDetails, [studentId]);
+  const abilityLength = [];
+  console.log(abilityLength);
 
-  const details = (skill) => {
+  const displayDetails = (skill) => {
     return studentDetail
       .filter((lo) => lo.skill === skill)
       .map(({ description, ability }) => {
         return (
           <p>
-            - {description}:<button>score :{ability}</button>
+            - {description}: <button> score :{ability} </button>
           </p>
         );
       });
   };
+
+  const getAverageAbility = (skill) => {
+    const filteredResults = studentDetail.filter((lo) => lo.skill === skill);
+    const totalAbility = filteredResults.reduce(
+      (currentTotalAbility, { ability }) => currentTotalAbility + ability,
+      0
+    );
+
+    const averageAbility = Math.round(
+      (totalAbility / (filteredResults.length * 2)) * 100
+    );
+
+    console.log(filteredResults);
+
+    if (averageAbility) {
+      return " " + averageAbility + " " + "%";
+    } else {
+      return " " + "Not covered yet!";
```

Comment:
If the user has selected "Not confident" for each answer, then the `averageAbility` will be 0. In this case, this will show "Not covered yet!". Not sure if this is what you expect here?